htmresearch/algorithms/simple_union_pooler.py

```diff
@@ -0,0 +1,131 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import numpy
+from nupic.bindings.math import GetNTAReal
+
+REAL_DTYPE = GetNTAReal()
+UINT_DTYPE = "uint32"
+
+
+class SimpleUnionPooler(object):
+  """
+  Experimental Simple Union Pooler Python Implementation.
+  The simple union pooler computes a union of the last N SDRs
+  """
+
+  def __init__(self,
+               inputDimensions=(2048,),
+               historyLength=10):
+    """
+    Parameters:
+    ----------------------------
+    @param inputDimensions:
+    A sequence representing the dimensions of the input vector. Format is
+    (height, width, depth, ...), where each value represents the size of the
+    dimension.  For a topology of one dimension with 100 inputs use 100, or
+    (100,). For a two dimensional topology of 10x5 use (10,5).
+
+    @param historyLength: The union window length. For a union of the last
+    10 steps, use historyLength=10
+    """
+
+    self._inputDimensions = inputDimensions
+    self._historyLength = historyLength
+    self._numInputs = 1
+    for d in inputDimensions:
+      self._numInputs *= d
+
+    self.reset()
+
+
+  def reset(self):
+    """
+    Reset Union Pooler, clear active cell history
+    """
+    self._unionSDR = numpy.zeros(shape=self._inputDimensions, dtype=UINT_DTYPE)
+    self._activeCellsHistory = []
+
+
+  def compute(self, activeCells):
```

Comment:
Just call this function updateHistory(self, activeCells) and don't create the union - it's not used anywhere!!